### PR TITLE
feat(sql): support window (union = (last join)) in online serving mode

### PR DESCRIPTION
**Pull request by aceforeverd**
```sql
select ..
from t
window w as (
   union (select .. from t2 last t3 on ...)
   partition by .. order by ...
)
```

-> 

```
AggNode
  RequestUnion
    Request(t)
    Table(t)
    WindowUnions
      SimpleProject
        RequestJoin
          Partition(t2)
          Partition(t3)
```

- t2 should optimized by window definition
- t3 should optimized by join condition

### CODE REVIEW

File: hybridse/src/vm/runner.cc [hunk 1]
```diff
@@ -347,9 +363,15 @@ ClusterTask RunnerBuilder::Build(PhysicalOpNode* node, Status& status) {
                         left->output_schemas()->GetSchemaSourceSize(), right->output_schemas()->GetSchemaSourceSize(),
                         op->output_right_only());
 
-                    return RegisterTask(
-                        node, BinaryInherit(left_task, right_task, runner,
-                                            op->join().index_key(), kLeftBias));
+                    if (IsPartitionProvider(node->GetProducer(0))) {
```
Comment (aceforeverd) [inline hybridse/src/vm/runner.cc]: the trick path

File: hybridse/src/vm/runner.cc [hunk 2]
```diff
@@ -648,23 +668,30 @@ ClusterTask RunnerBuilder::BuildClusterTaskForBinaryRunner(
             return ClusterTask();
         }
         if (right.IsCompletedClusterTask()) {
-            LOG(WARNING) << "Fail to complete cluster task for "
-                         << "[" << runner->id_ << "]"
-                         << RunnerTypeName(runner->type_)
-                         << ": task is completed already";
-            return ClusterTask();
+            if (node::ExprEquals(right.GetIndexKey().keys(), index_key.keys())) {
```
Comment (aceforeverd) [inline hybridse/src/vm/runner.cc]: this change should not necessary now

File: hybridse/src/vm/catalog_wrapper.h [hunk 3]
```diff
@@ -565,6 +562,151 @@ class RowCombineWrapper : public RowHandler {
...
+ public:
+    LazyLastJoinPartitionHandler(std::shared_ptr<PartitionHandler> left, std::shared_ptr<PartitionHandler> right,
+                                const Row& param, std::shared_ptr<JoinGenerator> join);
+    ~LazyLastJoinPartitionHandler() override {}
+
+    // NOTE: only support get segement by key from left source
+    std::shared_ptr<TableHandler> GetSegment(const std::string& key) override;
+
+    std::unique_ptr<WindowIterator> GetWindowIterator() override {
+        return left_->GetWindowIterator();
```
Comment (aceforeverd) [inline hybridse/src/vm/catalog_wrapper.h]: comment those interface that do not implemented yet

**Comment by aceforeverd** [comment]
TODO further: 
- [ ] lazy last join for batch / batch request mode
- [ ] more test cases
- [ ] window ( union = multiple join ) 
- [x] related doc update

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Pull request merged**